File: backend/api/src/main/java/com/github/ai/leetcodequiz/api/ProblemItemDto.java
package com.github.ai.leetcodequiz.api;

import org.jetbrains.annotations.NotNull;

import java.util.List;

public record ProblemItemDto(
    int id,
    @NotNull String title,
    @NotNull String content,
    @NotNull List<String> hints,
    @NotNull String categoryTitle,
    @NotNull String difficulty,
    @NotNull String url,
    int likes,
    int dislikes
) {
}

File: backend/api/src/main/scala/com/github/ai/leetcodequiz/api/ProblemItemDto.scala
package com.github.ai.leetcodequiz.api

import zio.json.{JsonDecoder, JsonEncoder}

case class ProblemItemDto(
  id: Int,
  title: String,
  content: String,
  hints: List[String],
  categoryTitle: String,
  difficulty: String,
  url: String,
  solutions: List[SolutionItemDto],
  questions: List[QuestionItemDto],
  likes: Int,
  dislikes: Int
) derives JsonEncoder,
      JsonDecoder
